### ISSUE TITLE
feature request: folding code to minimize footprint or distractions

### ISSUE THREAD

**Issue by frehov** [anboralabs/intellij-firebase-highlighter#47]
I would like to request the addition of code folding inside the rules-file.

I believe it would be beneficial to have such functionality if you have large and/or are implementing a large amount of custom functions to help validate.

We have a case where code folding would come in handy, and be able to collapse the entire match path for a collection.
We have - for now- 3 functions within the scope of the match path of a collection, and we are unit testing our rules, so after our tests pass there's no need to focus on that section anymore and it would be beneficial to be able to collapse it down again.

```
match /users/{user} {
  function isSelf() {
    return request.auth.uid == resource.id;
  }

  function isAllowedToReadAnotherUser() {
    let user = get(/databases/$(database)/documents/users/$(request.auth.uid));
    return userHasRole(user, ["role_1", "role_2"]);
  }

  function isAllowedToChangeAnotherUser() {
    let user = get(/databases/$(database)/documents/users/$(request.auth.uid));
    return userHasRole(user, ["role_2"]);
  }

  allow read: if isSignedIn() && (isSelf() || isAllowedToReadAnotherUser());
  allow write: if isSignedIn() && isAllowedToChangeAnotherUser();
}
```

the above is an example block which would have been nice to have to collapse into the following

```
match /users/{user} { ... }
```

I believe this reference should be of help. https://plugins.jetbrains.com/docs/intellij/folding-builder.html#define-a-folding-builder

**Comment by dalgarins** [comment]
<img width="414" alt="Captura de Pantalla 2021-02-14 a la(s) 7 59 53 a  m" src="https://user-images.githubusercontent.com/8527174/107877472-b83daa00-6e9a-11eb-8e07-43e35b33f817.png">

@frehov Thanks for comments, the feature will be available for version: 2.4.0

**Comment by frehov** [comment]
thanks for the update, this has really made my day so much easier when dealing with the firebase rules
